scripts/js/basic_usage.js

```diff
@@ -28,14 +28,49 @@ async function getBlockWithRetry(web3){
 
       i++;
       if (i === 10){
-        throw new Error('Failed to connect to Infura over websockets after 10 tries');
+        if (provider === 'infura') {
+          throw new Error('Failed to connect to Infura over websockets after 10 tries');
+        } else if (provider === 'alchemy') {
+          throw new Error('Failed to connect to Alchemy over websockets after 10 tries');
+        }
       }
 
     }
   }
   return block;
 }
 
+async function tryMultipleProviders(infuraProvider) {
+  try {
+    const web3 = new Web3(infuraProvider);
+    const block = await getBlockWithRetry(web3, 'infura');
+    return [
+      web3,
+      block
+    ];
+  } catch (error) {
+    if (infuraProvider.includes("wss")) {
+      // Fetched provider from ethers.js https://github.com/ethers-io/ethers.js/blob/50d712d6afc927aaee68b482157598554a45d403/packages/providers/src.ts/alchemy-provider.ts#L21
+      const alchemyProvider = "wss://eth-mainnet.alchemyapi.io/v2/_gg7wSSi0KMBsdKnGVfHDueq6xMB9EkC";
```

Comment:
are these ws and http public endpoints ? or are these API keys specifically created and maintained by ethers in alchemyapi ?